toolz/compatibility.py

```diff
@@ -5,6 +7,68 @@
 __all__ = ('PY3', 'map', 'filter', 'range', 'zip', 'reduce', 'zip_longest',
            'iteritems', 'iterkeys', 'itervalues', 'filterfalse')
 
+# This is a kludge for Python 3.4.0 support
+# currently len(inspect.getargspec(map).args) == 0, a wrong result.
+# As this is fixed in future versions then hopefully this kludge can be
+# removed.
+known_numargs = {map: 2, filter: 2, reduce: 2}
+
+
+def _num_required_args_p2(func):
+    """ Number of args for func
+
+    >>> def foo(a, b, c=None):
+    ...     return a + b + c
+
+    >>> _num_required_args_p2(foo)
+    2
+
+    >>> def bar(*args):
+    ...     return sum(args)
+
+    >>> print(_num_required_args_p2(bar))
+    None
+    """
+    if func in known_numargs:
+        return known_numargs[func]
+    try:
+        spec = inspect.getargspec(func)
+        if spec.varargs:
+            return None
+        num_defaults = len(spec.defaults) if spec.defaults else 0
+        return len(spec.args) - num_defaults           
+    except TypeError:
+        return None
+
+
+def _num_required_args_p3(func):
+    """ See description of _num_required_args_p2
+    """
+    if func in known_numargs:
+        return known_numargs[func]
+    try:
+        required_args = 0
+        for param in inspect.signature(func).parameters.values():
+            kind = param.kind
+            # Presence of *args or **kwargs precludes a requirement.
+            if kind in [param.VAR_POSITIONAL, param.VAR_KEYWORD]:
+                return None
+            # Keyword only arguments, and ones with a default aren't required.
+            elif kind == param.KEYWORD_ONLY or param.default != inspect._empty:
```

Comment:
Removing param.KEYWORD_ONLY from that line and adding it to the next elif would fix that case, ie it would return 1 required arg. Is that the desired behavior?